app/src/auto_validator/core/admin.py

```diff
@@ -94,26 +96,30 @@ class ValidatorInstanceAdmin(admin.ModelAdmin):
 
 @admin.register(Server)
 class ServerAdmin(admin.ModelAdmin):
-    list_display = ("name", "ip_address", "validatorinstance", "description", "created_at")
-    search_fields = ("name", "ip_address", "validator_instance__subnet_slot__subnet__name")
+    list_display = ("name", "ip_address", "subnet_slot", "validatorinstance_status", "description", "created_at")
+    search_fields = ("name", "ip_address", "validator_instances__subnet_slot__subnet__name")
 
-    def validatorinstance(self, obj):
-        return obj.validator_instances.hotkey if obj.validator_instances else "N/A"
+    def subnet_slot(self, obj):
+        return obj.validator_instances.subnet_slot if obj.validator_instances else "N/A"
```

Comment:
This will cause the ORM to issue a separate query for every individual object. You need to use `list_select_related = ('validator_instances', 'validator_instances__subnet_slot')` or something like it.

You should use django debug toolbar during development and you should have seen N queries being done during manual testing.